scenes/interact/interact_ray.gd

```diff
@@ -16,6 +15,12 @@ func _ready() -> void:
 		character = owner
 
 
+func interact_area() -> InteractArea:
```

Comment:
Either call this `get_interact_area()` or add a getter. Also `RayCast2D.get_collider()` already returns null if no object is intersecting the ray. So you should be able to do (untested!):

```
var interact_area: InteractArea:
	get:
		return get_collider() as InteractArea
```